packages/jpgwire/src/util.ts

```diff
@@ -109,45 +109,63 @@ export async function connectPgWire(
     connectionOptions.replication = 'database';
   }
 
-  let tlsOptions: tls.ConnectionOptions | false = false;
-
   if (config.sslmode != 'disable') {
     connectionOptions.sslmode = 'require';
-
-    tlsOptions = makeTlsOptions(config);
   } else {
     connectionOptions.sslmode = 'disable';
   }
+
   const connection = pgwire.pgconnection(connectionOptions as pgwire.PgConnectOptions);
+  patchConnection(connection, config);
 
-  // HACK: Not standard pgwire options
+  await connection.query();
```

Comment:
Is this call intentional?
I assume it would be to open/initialize the connection, but it would be good to have a comment here explaining this.

Reply:
It is intentional, `pgwire` has been changed to connect lazily. Running an empty `query()` is what the library does as well. I agree the "postgres query failed" error is not helpful, I've added a `catch` to recover the inner error in that case.

I've also added comments with links to the relevant sources in `pgwire`.